cadc-vos/src/main/java/ca/nrc/cadc/vos/TransferReader.java

```diff
@@ -317,4 +295,29 @@ else if (strDirection.equalsIgnoreCase(Direction.BIDIRECTIONAL.getValue()))
         }
         return rtn;
     }
+
+    private List<URI> parseTargets(Element root, Namespace vosNS) throws URISyntaxException {
+        List<URI> rtn = null;
+        List targs = root.getChildren("target", vosNS);
+
+        if (targs != null && targs.size() > 0) {
+            rtn = new ArrayList<URI>(targs.size());
+            for (Object obj : targs) {
+                Element eTarget = (Element) obj;
+                URI target = new URI(eTarget.getText());
```

Comment:
Maybe call validateTarget() here so there is only one loop?

Reply:
combined the validate step into parseTargets.

Reply:
Looks good